src/server/models/grid.js

```diff
@@ -19,40 +21,41 @@ class Grid {
     constructFrameList() {
         let frameList = [];
 
-        for (let i = 0; i < 10; i++) {
+        for (let i = 0; i < this.nbFrame; i++) {
             frameList.push(new Frame());
         }
 
         return frameList;
     }
 
-    calculFrame(frameTable, mancheNumber) {
+    calculFrame(namePlayer, mancheNumber) {
 
-        if (mancheNumber == 10) {
-            frameTable[mancheNumber - 1].setScore(calcFrame10(frameTable[mancheNumber - 1]));
+        frameTable=this.players[namePlayer];
+        if (mancheNumber == this.nbFrame) {
+            frameTable[mancheNumber - 1].setScore(calcFrame10(frameTable[mancheNumber - 1]),this.nbKeel);
         } else {
 
-            if (frameTable[mancheNumber - 1].getC1() == 10) { //case of a strike
+            if (frameTable[mancheNumber - 1].getC1() == this.nbKeel) { //case of a strike
 
 
-                if (frameTable[mancheNumber].getC1() == 10) { //case 2 strikes in a row
-                    if (mancheNumber == 9) { // case of the 9th frame with 2 strikes in a row
-                        frameTable[mancheNumber - 1].setScore(20 + frameTable[mancheNumber].getC2());
+                if (frameTable[mancheNumber].getC1() == this.nbKeel) { //case 2 strikes in a row
+                    if (mancheNumber == this.nbFrame-1) { // case of the 9th frame with 2 strikes in a row
+                        frameTable[mancheNumber - 1].setScore(this.nbKeel*2 + frameTable[mancheNumber].getC2(),this.nbKeel);
```

Comment:
Maybe add a parameter upper for this calcul

Reply:
done